mantis-runtime/src/main/java/io/mantisrx/runtime/executor/SinkPublisher.java

```diff
@@ -88,31 +87,29 @@ public void start(StageConfig<?, T> stage,
         final Observable wrappedO = merged.lift(new MonitorOperator("worker_sink"));
 
         Observable o = Observable
-                .create(new Observable.OnSubscribe<Object>() {
-                    @Override
-                    public void call(Subscriber subscriber) {
-                        logger.info("Got sink subscription, onSubscribe=" + onSubscribeAction);
-                        wrappedO
-                                .doOnCompleted(observableOnCompleteCallback)
-                                .doOnError(observableOnErrorCallback)
-                                .doOnTerminate(observableTerminatedCallback)
-                                .subscribe(subscriber);
-                        if (onSubscribeAction != null) {
-                            onSubscribeAction.call();
-                        }
+                .create(subscriber -> {
+                    logger.info("Got sink subscription with onSubscribeAction={}", onSubscribeAction);
+                    wrappedO
+                            .doOnCompleted(observableOnCompleteCallback)
+                            .doOnError(observableOnErrorCallback)
+                            .doOnTerminate(observableTerminatedCallback)
+                            .subscribe(subscriber);
+                    if (onSubscribeAction != null) {
+                        onSubscribeAction.call();
                     }
                 })
-                .doOnUnsubscribe(new Action0() {
-                    @Override
-                    public void call() {
-                        logger.info("Sink subscriptions clean up, action=" + onUnsubscribeAction);
-                        if (onUnsubscribeAction != null)
-                            onUnsubscribeAction.call();
-                    }
+                .doOnCompleted(() -> logger.info("Sink observable subscription completed."))
```

Comment:
nit: is other information useful in the log like subscribe etc.

Reply:
subscribe?